src/GraphManager/utils.test.ts

```diff
@@ -264,3 +264,15 @@ describe("linkPointerAreaPaint", () => {
     expect(ctx.stroke).not.toHaveBeenCalled();
   });
 });
+
+import { onGraphUpdate } from "./GraphRenderer";
+
+describe("onGraphUpdate", () => {
+  it("should update the graph data", () => {
+    const ctrl = makeMockController();
+    const graphData = { nodes: [], links: [] };
+    const state = makeGraphState(graphData, jest.fn());
```

Comment:
It seems `state` is not used here, you could just `const setGraph = jest.fn();` and pass setGraph directly to `onGraphUpdate`.